smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude-traits.smithy

```diff
@@ -148,7 +148,7 @@ structure xmlAttribute {}
 
 /// Moves serialized collection members from their collection element to that of
 /// the collection's container.
-@trait(selector: "collection")
+@trait(selector: ":each(collection, structure > :test(member > collection))")
```

Comment:
It would be simpler to do:

`:test(collection, member:of(structure) > collection)`